client_test.go

```diff
@@ -492,6 +492,46 @@ func Test_Client_Stop(t *testing.T) {
 	})
 }
 
+func Test_Client_Stopped(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	t.Run("ClosesWhenStopCompletes", func(t *testing.T) {
+		t.Parallel()
+
+		jobStartedChan := make(chan int64)
+		jobDoneChan := make(chan struct{})
+		callbackFunc := func(ctx context.Context, job *Job[callbackArgs]) error {
+			jobStartedChan <- job.ID
+			<-ctx.Done()
+			close(jobDoneChan)
+			return nil
+		}
+		client := runNewTestClient(ctx, t, newTestConfig(t, callbackFunc))
+
+		insertedJob, err := client.Insert(ctx, callbackArgs{}, nil)
+		require.NoError(t, err)
+
+		startedJobID := riverinternaltest.WaitOrTimeout(t, jobStartedChan)
+		require.Equal(t, insertedJob.ID, startedJobID)
+
+		select {
+		case <-client.Stopped():
+			t.Fatal("expected client to not be stopped yet")
+		default:
+		}
+
+		stopCtx, stopCancel := context.WithTimeout(ctx, 5*time.Second)
+		t.Cleanup(stopCancel)
+
+		if err := client.StopAndCancel(stopCtx); err != nil {
+			t.Fatal(err)
+		}
+
+		riverinternaltest.WaitOrTimeout(t, client.Stopped())
+	})
+}
```

Comment:
Probably not worth rewriting for this, but it might not be the worst idea to put this sort of thing in the general `Test_Client` suite for other changes of this sort. Normalizes code a little more, and needs less bespoke set up code for every client function.